sdk/core/azure-core/src/main/java/com/azure/core/http/HttpHeaderName.java

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http;
+
+import java.util.Locale;
+
+/**
+ * Declare various headers which can be used to interact with Azure services.
+ */
+public enum HttpHeaderName {
```

Comment:
It would be simpler to just have a string arg to `RetryPolicy` constructor.  This won't be necessary if you do that.

Reply:
See my answer on why I did not use String. But if  we all feel that  it is overengineering, we can just use String. I feel that header names are so common in rest  that somewhere in core we should have list of  all the  header names.. instead of using String in many libraries.